src/index.ml

```diff
@@ -736,6 +735,16 @@ struct
               may (fun lock -> IO.unlock lock) t.writer_lock ))
 
   let close = close' ~hook:(fun _ -> ())
+
+  let sync' ?hook t =
+    Stats.incr_nb_sync ();
+    let t = check_open t in
+    Log.info (fun l -> l "[%s] sync" (Filename.basename t.root));
+    if t.config.readonly then sync_log ?hook t else raise RW_not_allowed
+
+  let sync t = sync' t
```

Comment:
the `sync'` function doesn't seem necessary

Reply:
`sync` is hiding the hook used in `sync'` in the public API